service/controller/v11/resource/deployment/master_deployment.go

```diff
@@ -89,11 +90,15 @@ func newMasterDeployments(customObject v1alpha1.KVMConfig) ([]*extensionsv1.Depl
 				Template: apiv1.PodTemplateSpec{
 					ObjectMeta: apismetav1.ObjectMeta{
 						GenerateName: key.MasterID,
+						Finalizers: []string{
+							key.DrainingNodesFinalizer,
+						},
 						Labels: map[string]string{
-							"app":      key.MasterID,
-							"cluster":  key.ClusterID(customObject),
-							"customer": key.ClusterCustomer(customObject),
-							"node":     masterNode.ID,
+							"app":               key.MasterID,
+							"cluster":           key.ClusterID(customObject),
+							"customer":          key.ClusterCustomer(customObject),
+							"node":              masterNode.ID,
+							key.PodWatcherLabel: "kvm-operator",
```

Comment:
The label is used for the pod resource which is reconciled in its own framework. The deletion magic happens there.